src/scripts/main.js

```diff
@@ -1,5 +1,24 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
-});
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+if (!wall || !spider) {
+  // eslint-disable-next-line no-console
+  console.error('Wall or spider element not found!');
+} else {
+  wall.addEventListener('click', (e) => {
+    const wallCoords = wall.getBoundingClientRect();
+    const height = spider.clientHeight / 2;
+    const width = spider.clientWidth / 2;
+
+    let tops = e.clientY - wallCoords.top - wall.clientTop - height;
+    let lefts = e.clientX - wallCoords.left - wall.clientLeft - width;
```

Comment:
The calculations for `tops` and `lefts` should ensure that the spider's center is positioned under the cursor. The current logic subtracts half the spider's dimensions, which is correct for centering. Ensure that this logic is consistently applied to maintain the spider's center under the cursor.